zipline/pipeline/engine.py

```diff
@@ -751,6 +751,23 @@ def _validate_compute_chunk_params(self,
                     )
                 )
 
+    def adjust_date(self, pipeline, dt):
```

Comment:
Talked about this in person a bit. I think my preference here would be to split this into two piece of functionality:

1. `resolve_domain` (which already exists, but needs to be made public and tested)
2. `adjust_dates`, which can be part of the `Domain` interface.

The rationale for this split is twofold:

1. We might have consumers that have a domain and want to align dates to the domain's calendar, but don't want to construct a full-blown pipeline engine.
2. The domain is the logical "owner" of date information, so it feels like a more clean separation to have it be responsible for the rolling logic.

Reply:
made these changes, and split up the tests into test_domain.py and test_engine.py